firebase-perf/src/main/java/com/google/firebase/perf/gauges/CpuGaugeCollector.java

```diff
@@ -74,39 +76,41 @@
   // This utility isn't provided by TimeUnits.SECONDS.toMicros() - it only accepts longs.
   private static final long MICROSECONDS_PER_SECOND = TimeUnit.SECONDS.toMicros(1);
 
+  /* This is populated by CpuGaugeCollector but it's drained by GaugeManager.*/
+  public final ConcurrentLinkedQueue<CpuMetricReading> cpuMetricReadings =
+      new ConcurrentLinkedQueue<>();
+
+  // TODO: Remove sharedInstance.
   @Nullable private static CpuGaugeCollector sharedInstance = null;
   @Nullable private ScheduledFuture cpuMetricCollectorJob = null;
-
   private final ScheduledExecutorService cpuMetricCollectorExecutor;
   private long cpuMetricCollectionRateMs = UNSET_CPU_METRIC_COLLECTION_RATE;
   private final long clockTicksPerSecond;
   private final String procFileName;
 
-  /* This is populated by CpuGaugeCollector but it's drained by GaugeManager.*/
-  public final ConcurrentLinkedQueue<CpuMetricReading> cpuMetricReadings;
-
+  // TODO: Remove the private constructor.
   private CpuGaugeCollector() {
-    cpuMetricReadings = new ConcurrentLinkedQueue<>();
     cpuMetricCollectorExecutor = Executors.newSingleThreadScheduledExecutor();
 
     int pid = android.os.Process.myPid();
-    procFileName = "/proc/" + Integer.toString(pid) + "/stat";
+    procFileName = String.format("/proc/%d/stat", pid);
 
     clockTicksPerSecond = getClockTicksPerSecond();
   }
 
   @VisibleForTesting
+  @Inject
   CpuGaugeCollector(
       ScheduledExecutorService cpuMetricCollectorExecutor,
-      String fakeProcFileName,
-      long clockTicksPerSecond) {
-    cpuMetricReadings = new ConcurrentLinkedQueue<>();
+      @Named("proc file name") String procFileName,
```

Comment:
Prefer custom annotations to `@Named`, http://go/di-practices/prefer-custom-qualifiers

Reply:
Thanks for the reference! Done.